plugin-core/src/main/scala/com/alpine/plugin/core/OperatorParameters.scala

```diff
@@ -89,4 +89,5 @@ trait OperatorParameters {
    */
   def getParameterIds: Iterator[String]
 
+  def getAdvancedSparkParameters: scala.collection.mutable.Map[String, String]
```

Comment:
Switching this over to immutable would result in changes in 6 files. I need to take a deeper look first.